worker/docker_client.py

```diff
@@ -409,8 +345,17 @@ def check_finished(self, container_id):
             
             inspect_json = json.loads(inspect_response.read())
             if not inspect_json['State']['Running']:
-                return (True, inspect_json['State']['ExitCode'], None)
-            return (False, None, None)   
+                # If the logs are nonempty, then something might have gone
+                # wrong with the commands run before the user command,
+                # such as bash or cd.
```

Comment:
We do already check the exitcode elsewhere -- I added this in in order to capture errors that were previous hidden (for example the cryptic "Exit code 127" would have been easier to debug if we had been checking the logs of the container, which would have told us that the `sudo` command was not found).

However, I think I might remove this, now that I'm working on simplifying the setup commands (i.e. no more sourcing `bashrc`, or `ldconfig` and `stat`).